src/lib/y2network/autoinst/config.rb

```diff
@@ -46,11 +46,13 @@ class Config
       # @option opts [Integer] :ip_check_timetout
       # @option opts [Boolean] :virt_bridge_proposal
       def initialize(opts = {})
-        @before_proposal = opts.fetch(:before_proposal, false)
-        @start_immediately = opts.fetch(:start_immediately, false)
-        @keep_install_network = opts.fetch(:keep_install_network, true)
-        @ip_check_timeout = opts.fetch(:ip_check_timeout, -1)
-        @virt_bridge_proposal = opts.fetch(:virt_bridge_proposal, true)
+        ay_options = opts.delete_if { |_k, v| v.nil? }
```

Comment:
`#delete_if` modifies the original hash. Is that what you want? If not, you could use `#reject`.